src/components/institution/SelectInstitutionForm.tsx

```diff
@@ -80,12 +80,12 @@ export const SelectInstitutionForm = ({
         if (values.selectedSuggestedAffiliationId) {
           // When we select from a list of suggested affiliations
           saveAffiliation(values.selectedSuggestedAffiliationId);
-        } else if (values.subunit?.id) {
```

Comment:
Må flytte denne til etter sjekken for `selectedSubunitId`, ellers slå den inn også når man endret, sånn at endring bare førte til at samme verdi som før ble submitted igjen.

Må sjekke at dette ikke gir trøbbel noen andre plasser føre merge. @denektenina var den som jobbet med dette sist, så mulig du vet noe?

Reply:
Mulig problemet her egentlig er at `selectedSubunitId` har fått sin egen `useState`, som gjør at vi må passe på at den alltid er i sync med `values.subunit.id`?